pycoal/mineral.py

```diff
@@ -10,225 +10,79 @@
+        for x in xrange(M):
 
-            # TODO convert units ?
-            # see https://groups.google.com/d/msg/coal-capstone/6oordALy0dA/a_6VIuWbBAAJ
+            for y in xrange(N):
 
-            # define a linear transformation to scale the pixel if necessary
-            # the scalars would presumably be read from metadata
-            #transform = spectral.transforms.LinearTransform(???)
+                # read the pixel from the file
+                pixel = data[x,y]
 
-            # transform it
-            #transformedPixel = transform(pixel)
-
-            # classify and store it in the classified image
-            classifiedImage[imageIndex] = self.classifyPixel(pixel, classifier, wavelengthList)
-
-            imageIndex += 1
-        # save the classified image to a file
-        spectral.io.envi.save_classification(classifiedImageFilename, classifiedImage)
+                # if it is not a no data pixel
+                if not numpy.isclose(pixel[0], -0.005) and not pixel[0]==-50:
 
-    def classifyImages(self, imageFilenames, classifier, classifiedImageFilenames):
-        """
-        Classify a set of AVIRIS images using a trained classifier and save the results to files.
-        Args:
-           imageFilenames (str): the images to be classified
-           classifier (PerceptronClassifier): the classifier
-           classifiedImageFilenames (str): the files which to save the classified images
-        Returns:
-            None
-        """
+                    # resample the pixel ignoring NaNs from target bands that don't overlap
+                    # TODO fix spectral library so that bands are in order
+                    resampledPixel = numpy.nan_to_num(resample(pixel))
 
-        # for each image in a set of images
-        for imageFilename,classifiedImageFilename in zip(imageFilenames,classifiedImageFilenames):
+                    # calculate spectral angles
+                    angles = spectral.spectral_angles(resampledPixel[numpy.newaxis,
+                                                                     numpy.newaxis,
+                                                                     ...],
+                                                      self.library.spectra)
 
-            # classify the image and save it to a file
-            self.classifyImage(imageFilename, classifier, classifiedImageFilename)
+                    # get classification
+                    classified[x,y] = numpy.argmin(angles) + 1
 
+        # save the classified image to a file
+        spectral.io.envi.save_classification(classifiedFilename,
+                                             classified,
+                                             class_names=['No data']+self.library.names,
+                                             metadata={
+                                                 'data ignore value': 0,
+                                                 'description': 'Mineral classified image.',
```

Comment:
I don't think we do... you could try the following
```
import pkg_resources
pkg_resources.get_distribution("pycoal").version
```
If that doesn't work then maybe we can just create a constant somewhere.